Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8609,48 +8613,139 @@ def get_message_classes_dump(classes_as_list):
     for current_local_var in local_vars:
-        ret_value += str(current_local_var) + '\n'
+        ret_value += shorten_string_for_printing(str(current_local_var), 64) + '\n'
 
     ret_value += '\n--- End Local Vars ---\n\n'
 
     return ret_value
 
 
+def get_size_of_object(input_object):
+    """
+    A function that recursively iterate to sum size of object & members.
+
+    :type input_object: ``Any``
+    :param input_object: The object to calculate its memory footprint
+
+    :return: Size of input_object in bytes.
+    :rtype: ``int``
+    """
+    from collections import deque, Mapping
+    from numbers import Number
+    # Python 3 also needs range
+    # ZERO_DEPTH_BASES = (str, bytes, Number, range, bytearray)
+    ZERO_DEPTH_BASES = (str, bytes, Number, bytearray)
+    MAX_LEVEL = 20
+    _seen_ids = set()
+
+    def inner(obj, level):
+        if level == MAX_LEVEL:
+            # Stop at MAX_LEVEL
+            return sys.getsizeof(obj)
+        obj_id = id(obj)
+        if obj_id in _seen_ids:
+            return 0
+        _seen_ids.add(obj_id)
+        size = sys.getsizeof(obj)
+        if isinstance(obj, ZERO_DEPTH_BASES):
+            pass
+        elif isinstance(obj, (tuple, list, set, deque)):
+            size += sum(inner(i, level + 1) for i in obj)
+        elif isinstance(obj, Mapping):
+            mapping_items_keys = list(getattr(obj, 'keys')())
+            for current_key in mapping_items_keys:
+                if current_key in obj:
+                    size += (inner(current_key, level + 1) + inner(obj[current_key], level + 1))
+        # Check for custom object instances - may subclass above too
+        if hasattr(obj, '__dict__'):
+            size += inner(vars(obj), level + 1)
+        return size
+    return inner(input_object, 0)
+
+
+excluded_globals = ['__name__', '__doc__', '__package__', '__loader__',
+                    '__spec__', '__annotations__', '__builtins__',
+                    '__file__', '__cached__', '_Feature',
+                    ]
+excluded_types_names = ['MagicMock',  # When running tests locally
+                        ]
+
+
 def get_message_global_vars():
     """
-    A function that prints the global variables to log info
+    A function that returns the printable message about global variables
 
     :return: Message to print.
     :rtype: ``str``
     """
+    excluded_types = [types.ModuleType, types.FunctionType,
+                      ]
+
     globals_dict = dict(globals())
     globals_dict_full = {}
     for current_key in globals_dict.keys():
         current_value = globals_dict[current_key]
-        globals_dict_full[current_key] = {
-            'name': current_key,
-            'value': current_value,
-            'size': sys.getsizeof(current_value)
-        }
+        if not type(current_value) in excluded_types \
+                and current_key not in excluded_globals \
+                and type(current_value).__name__ not in excluded_types_names:
```

Comment:
are you using `type(...) in ` instead of `isinstance()` to exclude possible inheritance?

Reply:
This is how I got around `MagicMock` without needing to `include` it.

Reply:
I get it for `.__name__`, but that doesn't apply for `excluded_types = [types.ModuleType, types.FunctionType]` - in any case I don't think we should bother about classes that inherit from them, but it's good to keep this in mind.